CDTDatastore/CDTReplicator/CDTPullReplication.m

```diff
@@ -65,57 +65,17 @@ - (instancetype)copyWithZone:(NSZone *)zone
 
 - (NSString *)description
 {
-    NSMutableDictionary *dictionary = [[self dictionaryForReplicatorDocument:nil] mutableCopy];
-    dictionary[@"source"] = TDCleanURLtoString(self.source);
-    return [NSString stringWithFormat:@"%@: %@", [self class], dictionary];
-}
-
-- (NSDictionary *)dictionaryForReplicatorDocument:(NSError *__autoreleasing *)error
-{
-    NSError *localError;
-    if (![self validateRemoteDatastoreURL:self.source error:&localError]) {
-        if (error) {
-            *error = localError;
-        }
-        return nil;
-    }
-
-    NSDictionary *superdoc = [super dictionaryForReplicatorDocument:&localError];
-    if (superdoc == nil) {
-        if (error) {
-            *error = localError;
-        }
-        return nil;
-    }
-
-    NSMutableDictionary *doc = [NSMutableDictionary dictionaryWithDictionary:superdoc];
-
-    [doc setObject:self.source.absoluteString forKey:@"source"];
-
-    if (self.target) {
-        [doc setObject:self.target.name forKey:@"target"];
+    NSMutableDictionary *dictionary = [[NSMutableDictionary alloc] init];
```

Comment:
I think you need to revisit `description` for `CDTPullReplication` and `CDTPushReplication`.

These methods are only for debugging/logging etc, but I think they're no longer correct. You just need to build up a string which describes all of the options - source, target, headers, interceptors, filters, query_params etc.

Note that these methods previously relied in large part on `dictionaryForReplicatorDocument` to do this work and then printed a string representation of the dictionary.

My advice would be to dispense with the dictionary all together and just build up a string formatted something like this:
```
@"source: %@, target: %@, headers: %@, interceptors: %@, filter: %@, query_params: %@"
```
You don't have to worry about `nil` checks as these will just come out as `nil` in the formatted output.